### PR TITLE
Functional Tests : Migration TS (`@data/demo/*`)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : Migration TS (`@data/demo/*`)
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3959507602
| Fixed ticket?     | Relative to #30263 
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

**Comment by Progi1984** [comment]
Ping @boubkerbribri @nesrineabdmouleh @NeOMakinG 🤣

File: tests/UI/data/faker/linkWidget.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  public readonly staticPages: string[];
+
+  public readonly customPages: LinkWidgetPage[];
+
+  /**
+   * Constructor for class HookData
+   * @param valueToCreate {HookCreator} Could be used to force the value of some members
+   */
+  constructor(valueToCreate: LinkWidgetCreator = {}) {
+    this.name = valueToCreate.name || faker.word.noun();
```
Comment (boubkerbribri) [inline tests/UI/data/faker/linkWidget.ts:31]: Missing some js Doc here ?

Comment (Progi1984) [reply tests/UI/data/faker/linkWidget.ts:31]: Fixed

**Review by boubkerbribri — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**